### ISSUE TITLE
Number field over-restricts values

### ISSUE THREAD

**Issue by knod** [codeforboston/cliff-effects#475]
tl;dr: Now the field will only show valid numbers, even while you're typing them instead of waiting to return to a valid number after you're done.

### Replicate:
1. Click in a number field
2. Enter a number
3. Click out of the number field
4. Click back into the number field by any decimal
5. Try to edit the decimals
  a. Delete old numbers
  b. Insert a new number at a decimal place
  c. Select the whole number and replace it with another number

### Result I'd Expect:
a. If deleting decimals, to reduce the number of decimals until the decimal point itself is deleted.
b. If inserting a new number, for the number I've inserted to appear, lengthening the count of the numbers past the decimal point.
c. If selecting and replacing, to see one number without any decimal places.

### Actual Result:
a. The last number just turns into a 0, while the tenths place turns into the number that was at the hundredths place.
b. If at the last decimal place, usually a different number appears than what I've typed in. If at the first decimal place, the previous number there is pushed into the hundredths place.
c. A whole number with two decimal places that are both 0 is shown.

### Environment:
Operating system and version: OSX 10.11
Browser type and version: Chrome v65.0.3325.181

App version or commit id at the time: 9145d7ed8d881a5b63d71261378ec8736df0bcbf

### Other Notes
This may have happened because the changes made to create a blank field on a 0 value.
